src/utilities/downloads_backend.jl

```diff
@@ -6,22 +6,23 @@ using HTTP.MessageRequest: body_was_streamed
 This backend uses the Downloads.jl stdlib to use libcurl
 as an HTTP client to connect to the AWS REST API.
 
-It has one field,
+It has two fields,
 
-- `downloader::Union{Nothing,Downloads.Downloader}`
-
-which is the `Downloads.Downloader` to use. If set to `nothing`, the default,
-then a global downloader object will be used.
+- `downloader::Union{Nothing,Downloads.Downloader}`: if `nothing`, use a global Downloader object. Otherwise, uses the given Downloader.
+- `create_new_downloader::Any`: a zero-argument function which returns a new Downloader object to use.
+  Defaults to creating a new global downloader. This is called when a transient error occurs.
 
 Downloads.jl tends to perform better under concurrent operation than HTTP.jl,
 particularly with `@async` / `asyncmap`. As of March 2022, threading (e.g. `@spawn` or `@threads`) with Downloads.jl is broken on all releases of Julia ([Downloads.jl#110](https://github.com/JuliaLang/Downloads.jl/issues/110)), and there are still reported issues on the upcoming
 1.7.3 and 1.8 releases ([Downloads.jl#182](https://github.com/JuliaLang/Downloads.jl/issues/182])).
 """
 struct DownloadsBackend <: AWS.AbstractBackend
     downloader::Union{Nothing,Downloads.Downloader}
+    create_new_downloader::Any
 end
 
-DownloadsBackend() = DownloadsBackend(nothing)
+DownloadsBackend() = DownloadsBackend(nothing, () -> get_downloader(; fresh=true))
+DownloadsBackend(D::Downloader) = DownloadsBackend(D, () -> get_downloader(; fresh=true))
 
 const AWS_DOWNLOADER = Ref{Union{Nothing,Downloader}}(nothing)
```

Comment:
Not to derail this PR but using undef when the downloader is not yet defined seems preferable
```suggestion
const AWS_DOWNLOADER = Ref{Downloader}()
```
You'd just need to change some code in `get_downloader` to use `isassigned(AWS_DOWNLOADER)` before dereferencing

Reply:
Let's do that separately if we are going to change it. I vaguely remember some problems with that approach when trying it in the original implementation. Maybe it was useful to be able to reset back to nothing, or something like that...